server/src/main/java/org/elasticsearch/index/mapper/MappingLookup.java

```diff
@@ -42,14 +42,7 @@
     private final int metadataFieldCount;
     private final FieldNameAnalyzer indexAnalyzer;
 
-    private static void put(Map<String, Analyzer> analyzers, String key, Analyzer value, Analyzer defaultValue) {
-        if (value == null) {
-            value = defaultValue;
-        }
-        analyzers.put(key, value);
-    }
-
-    public static MappingLookup fromMapping(Mapping mapping, Analyzer defaultIndex) {
+    public static MappingLookup fromMapping(Mapping mapping) {
```

Comment:
We don't need the default analyzer any more because text-based field mappers now always provide an analyzer even if they're using a default.